src/main/java/com/avispl/dal/communicator/cisco/CiscoCommunicator.java

```diff
@@ -597,6 +597,123 @@ public void unmute() throws Exception {
+     */
+    private void routeMediaChannelsData(CiscoStatus ciscoStatus, EndpointStatistics endpointStatistics) {
+            MediaChannels mediaChannels = ciscoStatus.getMediaChannels();
+
+            if (mediaChannels == null) {
+                // this may mean that we are going legacy route since channel data is in the different part of payload in this case
+                populateCallChannelsData(ciscoStatus, endpointStatistics);
+            } else {
+                populateMediaChannelsData(ciscoStatus, endpointStatistics);
+            }
+        }
+
+    /**
+     * Populate media channels information, retrieved from the Call statistics of the device
+     *
+     * @param ciscoStatus to retrieve statistics from
+     * @param endpointStatistics to save data to
+     *
+     * @since 1.1.0
+     */
+    private void populateCallChannelsData(CiscoStatus ciscoStatus, EndpointStatistics endpointStatistics) {
+            Call[] calls = ciscoStatus.getCalls();
+            if (calls == null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Unable to populate media channels data: no calls information is available");
+                }
+                return;
+            }
+
+            List<Call> connectedCalls = Arrays.stream(ciscoStatus.getCalls()).filter(call -> "Connected".equals(call.getStatus()) || "Synced".equals(call.getStatus()))
+                    .collect(Collectors.toList());
+            long callsCount = connectedCalls.size();
+            if (callsCount > 1) {
+                throw new IllegalStateException(String.format("Ambiguous active calls found: %s, 1 expected. Unable to proceed.",
+                        callsCount));
+            }
+            if (callsCount == 0) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Unable to populate media channels data: no active calls is available");
+                }
+                return;
+            }
+            Call activeCall = connectedCalls.get(0);
+            Channel[] callChannels = activeCall.getChannels();
+            if (callChannels == null || callChannels.length == 0) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Unable to populate media channels data: no media channels data is available in the calls structure");
+                }
+                return;
+            }
+
+            AudioChannelStats audioChannelStats = new AudioChannelStats();
+            VideoChannelStats videoChannelStats = new VideoChannelStats();
+            CallStats callStats = new CallStats();
+            callStats.setCallId(activeCall.getCallId());
+            callStats.setRemoteAddress(activeCall.getRemoteNumber());
+            callStats.setProtocol(activeCall.getProtocol());
+
+            Arrays.stream(callChannels).forEach(channel -> {
+                String direction = channel.getDirection();
+                Audio[] audioChannels = channel.getAudio();
```

Comment:
No null check is necessary for both of these objects? I see them being used below, and making sure it won't cause NPE.